File: aat-android/src/main/java/ch/bailu/aat/views/html/HtmlScrollTextView.java
package ch.bailu.aat.views.html;

import android.content.Context;

import ch.bailu.aat.util.ui.UiTheme;
import ch.bailu.aat.views.preferences.VerticalScrollView;


public class HtmlScrollTextView extends VerticalScrollView {
    private final HtmlTextView textView;

    public HtmlScrollTextView(Context context) {
        super(context);

        textView = new HtmlTextView(context);
        add(textView);
    }

    public HtmlScrollTextView(Context context, String text, LinkHandler linkHandler) {
        this(context);
        setHtmlText(text);
        textView.setLinkHandler(linkHandler);
    }


    public HtmlScrollTextView(Context context, String text) {
        this(context);
        setHtmlText(text);
    }


    public void setHtmlText(String text) {
        textView.setHtmlText(text);
    }


    public void enableAutoLink() {
        textView.enableAutoLink();
    }

    public HtmlTextView getTextView() {
        return textView;
    }

    @Override
    public void setOnClickListener(OnClickListener l) {
        textView.setOnClickListener(l);
        super.setOnClickListener(l);
    }

    public void themify(UiTheme theme) {
        theme.content(textView);
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/html/HtmlScrollTextView.kt
package ch.bailu.aat.views.html

import android.content.Context
import ch.bailu.aat.util.ui.theme.UiTheme
import ch.bailu.aat.views.preferences.VerticalScrollView

class HtmlScrollTextView(context: Context, text: String = "") : VerticalScrollView(context) {
    val textView: HtmlTextView = HtmlTextView(context)

    init {
        add(textView)
        setHtmlText(text)
        textView.setTextIsSelectable(true)
    }

    fun setHtmlText(text: String) {
        textView.setHtmlText(text)
    }

    override fun setOnClickListener(listener: OnClickListener?) {
        textView.setOnClickListener(listener)
        super.setOnClickListener(listener)
    }

    fun themify(theme: UiTheme) {
        theme.content(textView)
    }
}
